rust/connlib/shared/src/lib.rs

```diff
@@ -68,7 +67,7 @@ pub fn get_user_agent(os_version_override: Option<String>) -> String {
 
     let os_version = os_version_override.unwrap_or(info.version().to_string());
     let additional_info = additional_info();
-    let lib_version = VERSION;
+    let lib_version = option_env!("FIREZONE_PACKAGE_VERSION").unwrap_or(env!("CARGO_PKG_VERSION"));
```

Comment:
when will this not exist? when compiling locally?

Reply:
for the lib name we could instead use `CARGO_BIN_NAME` so that we distinguish between all the binaries https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

Reply:
Yeah, when compiling locally. The problem with `CARGO_BIN_NAME` though is that it won't apply to the Apple/Android packages since they're `lib`s right?

Reply:
Added a corresponding `FIREZONE_PACKAGE_NAME` that gives us control over what the reported name will be.

Reply:
> Yeah, when compiling locally. The problem with CARGO_BIN_NAME though is that it won't apply to the Apple/Android packages since they're libs right?

hm, maybe `cdylib`s counts as binaries, but I'm not sure, we would need to test it.

Reply:
maybe instead of the crate version we could use `development` as the version for locally compiled? that way we can detect that case and we don't have confusions about the versions.

Reply:
Updated to stick with `connlib/` since the portal expects this (and we send the OS version anyway). Also updated to use `development` if FIREZONE_PACKAGE_VERSION is missing.